scripts/docs-test.js

```diff
@@ -0,0 +1,191 @@
+/*******************************************************************************
+
+    publicsuffixlist.js - an efficient javascript implementation to deal with
+    Mozilla Foundation's Public Suffix List <http://publicsuffix.org/list/>
+
+    Copyright (C) 2013-present Raymond Hill
+
+    License: pick the one which suits you:
+      GPL v3 see <https://www.gnu.org/licenses/gpl.html>
+      APL v2 see <http://www.apache.org/licenses/LICENSE-2.0>
+
+*/
+
+// Script for running the same tests as docs/test.html until there is a proper
+// Mocha-based test suite.
+
+import { readFileSync } from 'fs';
+import { domainToASCII } from 'url';
+
+import publicSuffixList from '../publicsuffixlist.js';
+
+let errorCount = 0;
+
+function customFetch(fileURL) {
+    const buffer = readFileSync(fileURL);
+    return ({
+        async arrayBuffer() {
+            return new Uint8Array(buffer).buffer;
+        }
+    });
+}
+
+const resultTemplateGood = '   Success: checkPublicSuffix({{a}}, {{b}})';
+const resultTemplateBad = '\u2757 Failure: got {{r}} instead: checkPublicSuffix({{a}}, {{b}})';
+
+function checkPublicSuffix(a, b) {
+    // publicSuffixList() does not normalize Unicode, it is the caller's
+    // responsibility, because overhead etc.
+    const anorm = a ? domainToASCII(a) : a;
+    const bnorm = b ? domainToASCII(b) : b;
+
+    let r = publicSuffixList.getDomain(anorm);
+    if ( !r ) {
+        r = null;
+    }
+    const template = r === bnorm ? resultTemplateGood : resultTemplateBad;
+    const output = template
+        .replace('{{a}}', a ? "'" + a + "'" : 'null')
+        .replace('{{b}}', b ? "'" + b + "'" : 'null')
+        .replace('{{r}}', r ? "'" + r + "'" : 'null');
+    console.log(output);
+
+    if ( template === resultTemplateBad ) {
+        errorCount++;
+    }
+}
+
+// This function is a verbatim copy of the one in docs/test.html
```

Comment:
This is an exact copy of the code in `docs/test.html`. It should be easy to keep them in sync. In the end I expect that this file will become redundant as we have better tests in `tests/*.js` eventually.